village/src/components/Smurf.js

```diff
@@ -1,13 +1,51 @@
 import React from 'react';
+import EditSmurf from './EditSmurf'
 
-const Smurf = props => {
-  return (
-    <div className="Smurf">
-      <h3>{props.name}</h3>
-      <strong>{props.height} tall</strong>
-      <p>{props.age} smurf years old</p>
-    </div>
-  );
+class Smurf extends React.Component{
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      edit: false
+    }
+  }
+
+  edit = (edit) => {
+    let id = this.props.id
+    this.props.edit(id, edit)
+    this.setState({edit: false});
+  }
+
+  clickHandler = () => {
+    if (this.state.edit) {
+      this.setState({edit: false})
+    } else{
+      this.setState({edit: true})
+    }
+  }
+
+  Form = () => {
```

Comment:
This could be done a _little_ differently.

You don't really need this as a function. Look below